src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -1621,19 +1625,21 @@ static void FillCheckEventInfo(Eventinfo *lf, cJSON *scan_id, cJSON *id, cJSON *
             char *value = NULL;
             int free_value = 0;
 
-            if(!comp->valuestring){
-                if(comp->valueint) {
-                    os_calloc(OS_SIZE_1024, sizeof(char), value);
-                    sprintf(value, "%d", comp->valueint);
-                    free_value = 1;
-                } else if(comp->valuedouble) {
-                    os_calloc(OS_SIZE_1024, sizeof(char), value);
-                    sprintf(value, "%lf", comp->valuedouble);
-                    free_value = 1;
-                }
-            } else {
+            if (cJSON_IsString(comp))
+            {
                 value = comp->valuestring;
             }
+            else if (cJSON_IsNumber(comp))
```

Comment:
Please, let's keep the typical format we use in the rest of the functions, with the bracket on the same line as the header.